pycqed/measurement/optimization.py

```diff
@@ -135,3 +135,99 @@ def nelder_mead(fun, x0,
     # verification
     fun(res[0][0])
     return res[0]
+
+
+def SPSA(fun, x0,
+         initial_step=0.1,
+         no_improve_thr=10e-6, no_improv_break=10,
+         maxiter=0,
+         gamma=0.101, alpha=0.602, a=0.2, c=0.3, A=300,
+         p=0.5,
+         verbose=False):
+    '''
+    parameters:
+        fun (function): function to optimize, must return a scalar score
+            and operate over a numpy array of the same dimensions as x0
+        x0 (numpy array): initial position
+
+
+        no_improv_thr,  no_improv_break (float, int): break after
+            no_improv_break iterations with an improvement lower than
+            no_improv_thr
+        maxiter (int): always break after this number of iterations.
+            Set it to 0 to loop indefinitely.
+        alpha, gamma, a, c, A: parameters for the SPSA gains
```

Comment:
@elrama- could you add the expected type ('m assuming float) to the docstring of these args? (better to be consistent with the rest.